### PR TITLE
Version file

### PR DESCRIPTION
**Pull request by haircommander**
To be able to set auto-upgrades, we need the ability write the current CRI-O version to a file
Add this to version/version.go
Add package google/renameio for an atomic write of the version file
Call WriteVersionFile in main, where we have the gitCommit and we've just created the server

forward-port of: https://github.com/cri-o/cri-o/pull/2397
Signed-off-by: Peter Hunt pehunt@redhat.com

### CODE REVIEW

**Comment by haircommander** [comment]
@mrunalp @umohnani8 @rhatdan PTAL

File: version/version.go [hunk 1]
```diff
@@ -1,4 +1,58 @@
...
+	"path/filepath"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/google/renameio"
+)
+
 // Version is the version of the build.
-const Version = "1.14.0-dev"
+const Version = "1.14.2-dev"
```
Comment (umohnani8) [inline version/version.go]: We cut 1.14.2, so might want to change this.

Comment (haircommander) [reply version/version.go]: hm, do we even update the version file in master as we cut branches? I think this change isn't supposed to be here...

Comment (umohnani8) [reply version/version.go]: My bad actually, we cut in https://github.com/cri-o/cri-o/pull/2435 but it hasn't gone in yet. And yeah, master keeps track of the `-dev` versions, so it should be there.

Comment (haircommander) [reply version/version.go]: so we can resolve this?

**Pull request merged**